pages/exercises/[lessonSlug].tsx

```diff
@@ -208,24 +248,39 @@ const ExerciseList = ({
         />
       </div>
       <div className="d-flex flex-column flex-md-row justify-content-between align-items-center">
-        <h1 className="my-2 my-md-5 fs-2">{lessonTitle}</h1>
-        <div
-          className={`mb-3 mb-md-0 d-flex d-md-block ${styles.exerciseList__solveExercisesButtonContainer}`}
-        >
-          <NewButton
-            className="flex-grow-1"
-            onClick={() => setExerciseIndex(0)}
-          >
-            SOLVE EXERCISES
-          </NewButton>
+        <div className="my-2 my-md-5">
+          <h1 className="fs-2">{lessonTitle}</h1>
+          <label className="d-inline-flex align-items-center">
+            <input
+              className="form-check-input m-0 me-3"
+              type="checkbox"
+              style={{ width: 30, height: 30 }}
+              checked={hideAnswered}
+              onChange={() => setHideAnswered(!hideAnswered)}
+            />
+            <span>Show incomplete exercises only</span>
+          </label>
         </div>
+        {exercises.length > 0 && (
+          <div
+            className={`mb-3 mb-md-0 d-flex d-md-block ${styles.exerciseList__solveExercisesButtonContainer}`}
+          >
+            <NewButton className="flex-grow-1" onClick={onClickSolveExercises}>
+              SOLVE EXERCISES
+            </NewButton>
+          </div>
+        )}
       </div>
       <div className={styles.exerciseList__container}>
         {exercises.map((exercise, i) => (
           <ExercisePreviewCard
             key={i}
             moduleName={exercise.moduleName}
-            state={exercise.userAnswer === null ? 'NOT ANSWERED' : 'ANSWERED'}
+            state={(() => {
+              if (exercise.userAnswer === exercise.answer) return 'ANSWERED'
```

Comment:
Instead of setting the state in the client side, is it better to store it in the user's `exerciseSubmission`?

```js
{
  ....,
  state: 'ANSWERED'
}
```

Reply:
Yeah, you're right. It's better to do all the data processing in one location. Also, if we need this property in multiple places in the future, it will be easier to grab it.

Reply:
Done.